### PR TITLE
Notification manager version 1 code

### PR DESCRIPTION
**Pull request by piyush-soni**

### CODE REVIEW

File: js/nm_notice_data.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+					|| all_notices[i].classList.contains("error")
+					|| all_notices[i].hasAttribute('aria-hidden'))
+					{continue;}
+				
+				nm_add_to_container(all_notices[i]);
+			}
+		}
+		else
+		{
+			document.getElementById("wpfooter").innerHTML += '<style>.notice {visibility:unset;}.updated {visibility:unset;}.error {visibility:unset;}</style>';
```
Comment (bv-ankit) [inline js/nm_notice_data.js:65]: Take a review of shivam sir on these things once.

File: js/nm_notice_data.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+			nm_sanitize(document.querySelectorAll(".notice.notice-success"));
+			nm_sanitize(document.querySelectorAll(".updated"));
+			nm_sanitize(document.querySelectorAll(".notice.notice-warning"));
+			nm_sanitize(document.querySelectorAll(".notice.notice-info"));
+
+			var all_notices = document.querySelectorAll(".notice");
+
+			for(var i=0; i<all_notices.length; i++)
+			{
+				if(all_notices[i].classList.contains("notice-success")
```
Comment (bv-ankit) [inline js/nm_notice_data.js:50]: Refactor

File: js/nm_notice_data.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+			var nm_div_box = document.querySelector(nm_notice_div_id);
+
+			nm_notice.classList.add("inline","notice-alt");
+			nm_notice.style.visibility = "unset";
+			
+			nm_div_box.appendChild(nm_notice);
+			notices_id++;
+		}
+
+		function nm_sanitize(nm_notices)
```
Comment (bv-ankit) [inline js/nm_notice_data.js:27]: function does more than what its name suggests

**Review by bv-ankit — CHANGES_REQUESTED** [review]

File: notification-manager.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+			'title' => '<div style="text-align:right"> Notices </div>',
+			'href' => false,
+			'meta' => array('onclick' => '{var x = document.getElementById("nm_container"); x.style.visibility = x.style.visibility === "visible" ? "hidden" : "visible";}'),
+			)
+		);
+
+		echo '<div id="nm_container"><h3> There is no notification to display </h3></div>';
+	}
+
+	add_action('init', 'nm_main');
```
[the remark this answers is not in the document]
Comment (piyush-soni) [reply notification-manager.php:52]: init is kind of a standard practice for adding new functions to a plugin, there are other options also like wp_loaded but init can also be used. so i am choosing this way of including my php function

**Pull request closed without merging**